### DISCUSSION TITLE
Custom Expect function, getting source code related to spec file in Reporter

### DISCUSSION THREAD

**Discussion by jkavalec** [vitest-dev/vitest#5331] [Q&A] [answered]
![image](https://github.com/vitest-dev/vitest/assets/15125854/249f2111-964e-4a8a-abbb-34d3d123cef7)

I am working on some Test matchers, and want to have them working in Vitest, there is one last thing I need to solve.

I need reporter to display Source code related to spec and not Source of custom Expect function, I throw an AssertionError, which gets thrown by standard expect/matcher combo, yet my custom Expect has code shown for its code and not test code.

A good thing is that information regarding spec file for the default matcher for instance, is already there - as shown on screenshot, the very last statement shows exact location that I would want to have displayed, but instead custom Expect code is shown.

From informations I have I know there will be possibilities to make it work, however I would prefer some very simple solution, as all the data are clearly within default Exporter's reach, without implementing custom Reporter, and getting more involved with vitest internals0

Is there any way how to make Vitest ignore custom Expect error origin, and do the same thing as Vitest own expect?

Or is there any good info how the mechanism of Vitest Reporter getting the source pretty printed from knowing code location work?

**Comment by sheremet-va** [comment, marked as answer]
You can use [`Error.captureStackTrace`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error#error.capturestacktrace) to remove the current function from the stack:

```js
export const match = (received, expected) => {
  const error = new Error('Error message')
  Error.captureStackTrace(error, match)
  throw error
}
```

**Reply by jkavalec** [reply]
Thank you very much, this solved it perfectly.
